src/client/Item/Item.ts

```diff
@@ -135,12 +124,16 @@ export class Item {
    *
    * @param options Additional options for the request, such as the partition key.
    */
-  public delete<T extends ItemDefinition>(options?: RequestOptions): Promise<ItemResponse<T>>;
-  public async delete<T extends ItemDefinition>(options?: RequestOptions): Promise<ItemResponse<T>> {
-    options = options || {};
-    if ((!options || !options.partitionKey) && this.partitionKey !== undefined) {
-      options.partitionKey = this.partitionKey;
+  public async delete<T extends ItemDefinition = any>(options: RequestOptions = {}): Promise<ItemResponse<T>> {
+    if (!options.partitionKey) {
+      if (this.partitionKey !== undefined) {
+        options.partitionKey = this.partitionKey;
+      } else {
+        const { resource: partitionKeyDefinition } = await this.container.getPartitionKeyDefinition();
+        options.partitionKey = extractPartitionKey(undefined, partitionKeyDefinition);
```

Comment:
It seems really weird to pass an undefined body to this. Shouldn't this always be this.partitionKey? Couldn't you replace this whole if/else block with something like:

```typescript
options.partitionKey = this.partitionKey !== undefined 
                                      ? this.partitionKey 
                                      : (partitionKeyDefintion.IsSystem ? [] : [{}]);
```

Reply:
Nested ternaries? You are a madman

But yes.